crates/sui/src/ptb/ptb.rs

```diff
@@ -339,22 +340,18 @@ impl PTB {
 
         // Build the PTB
         let mut parser = PTBParser::new();
-        for command in commands {
+        for command in commands.clone() {
             parser.parse(command.1);
         }
 
         let (parsed, errors) = parser.finish();
 
         if !errors.is_empty() {
-            let errs: Vec<_> = errors
-                .iter()
-                .enumerate()
-                .map(|(i, e)| format!("[{i}]: {e}"))
-                .collect();
-            anyhow::bail!(
-                "Encountered errors when parsing the PTB\n{}",
-                errs.join("\n")
-            );
+            let rendered = render_errors(commands, errors);
+            for e in rendered.iter() {
+                println!("{:?}", e);
```

Comment:
In the long run we probably should impl Display for errors so that we don't have to print their debug version.

Reply:
Yea, it's kinda weird -- the `miette` errors don't implement display, but the debug display gives the display you'd want/expect from `Display`. There has to be a reason I'm sure, but it definitely feels a touch odd...

Reply:
Interesting, did not know that. Then no worries here :D